File: generators/web/prototype/org.obeonetwork.informationsystem.gen.web.proto/asset/js/app/Root.js
/*******************************************************************************
 * Copyright (c) 2012 Obeo.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 * 
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/

/*
 * @autor hugo.marchadour@obeo.fr
 * @version 0.2
 * 
 */
define(["require", "app/App", "app/ctrl/commonCtrl", "app/ctrl/homeCtrl", "bs", "ember"], function(require) {
	
	var App = require("app/App");
	var commonCtrl = require("app/ctrl/commonCtrl");
	var homeCtrl = require("app/ctrl/homeCtrl");
	
	/* Initialize all sub controllers
	 */
	function _init() {
		
		
		
		App.debug = true;
		App.commonCtrl = commonCtrl;
		App.homeCtrl = homeCtrl;
		
		App.commonCtrl.start();
		App.homeCtrl.start();

	}

	function start() {
		_init();
	}

	start();
});

File: models/overview/plugins/org.obeonetwork.dsl.overview/src/org/obeonetwork/dsl/overview/Root.java
/*******************************************************************************
 * Copyright (c) 2008, 2017 Obeo.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/
package org.obeonetwork.dsl.overview;

import org.eclipse.emf.cdo.CDOObject;
import org.eclipse.emf.common.util.EList;
import org.eclipse.emf.ecore.EObject;
import org.obeonetwork.dsl.environment.ObeoDSMObject;

/**
 * <!-- begin-user-doc -->
 * A representation of the model object '<em><b>Root</b></em>'.
 * <!-- end-user-doc -->
 *
 * <p>
 * The following features are supported:
 * </p>
 * <ul>
 *   <li>{@link org.obeonetwork.dsl.overview.Root#getOwnedElements <em>Owned Elements</em>}</li>
 * </ul>
 *
 * @see org.obeonetwork.dsl.overview.OverviewPackage#getRoot()
 * @model
 * @generated
 */
public interface Root extends ObeoDSMObject {
	/**
	 * Returns the value of the '<em><b>Owned Elements</b></em>' containment reference list.
	 * The list contents are of type {@link org.eclipse.emf.ecore.EObject}.
	 * <!-- begin-user-doc -->
	 * <p>
	 * If the meaning of the '<em>Owned Elements</em>' containment reference list isn't clear,
	 * there really should be more of a description here...
	 * </p>
	 * <!-- end-user-doc -->
	 * @return the value of the '<em>Owned Elements</em>' containment reference list.
	 * @see org.obeonetwork.dsl.overview.OverviewPackage#getRoot_OwnedElements()
	 * @model containment="true" resolveProxies="true"
	 * @generated
	 */
	EList<EObject> getOwnedElements();

} // Root
